WordPress/gutenberg
# Update help text alignment in CheckboxControl

## What?
Update the margin applied to help text in checkbox control so that it aligns with the label.

## Why?
improves readability, and aligns with the treatment of status radios in the following UI:

<img width="316" alt="Screenshot 2024-04-16 at 16 41 52" src="https://github.com/WordPress/gutenberg/assets/846565/30f61596-1efb-4e41-a84e-2e872e7d9032">


## Testing Instructions
1. Run `npm run storybook:dev`
2. Navigate to CheckboxControl and check the text alignment

1. Open the site editor
2. Navigate to the details page for the "Blog Home" template (you may need to create this template)
3. Observe the updated text alignment in the "Discussion" settings. 

### Before
<img width="202" alt="Screenshot 2024-04-16 at 16 34 51" src="https://github.com/WordPress/gutenberg/assets/846565/4b142512-e7b3-4d8e-8077-87ec6d0b62ff">
<img width="356" alt="Screenshot 2024-04-16 at 16 39 11" src="https://github.com/WordPress/gutenberg/assets/846565/fd1d9801-1852-4320-8601-3e55f6c66366">

### After
<img width="245" alt="Screenshot 2024-04-16 at 16 34 57" src="https://github.com/WordPress/gutenberg/assets/846565/fe7699d4-3da2-4924-a52e-3e56d3a827e0">
<img width="354" alt="Screenshot 2024-04-16 at 16 39 33" src="https://github.com/WordPress/gutenberg/assets/846565/bd2433ad-379b-44cd-9514-f5e98be32016">
